common/browser_process.go

```diff
@@ -179,41 +179,51 @@ func execute(
 	return command{cmd, done, stdout, stderr}, nil
 }
 
-// parseDevToolsURL grabs the websocket address from chrome's output and returns it.
-func parseDevToolsURL(ctx context.Context, cmd command) (wsURL string, _ error) {
-	type result struct {
-		devToolsURL string
-		err         error
-	}
-	c := make(chan result, 1)
+// parseDevToolsURL grabs the WebSocket address from Chrome's output and returns
+// it. If the process ends abruptly, it will return the first error from stderr.
+func parseDevToolsURL(ctx context.Context, cmd command) (string, error) { //nolint: cyclop
```

Comment:
> a slight increase in cyclomatic complexity
> the current version is more readable

TBH, I had a hard time trying to understand it. Can you simplify the function further?

I'm also not comfortable adding another `nolint` (#58) while there is a chance for further refactoring.

Reply:
> TBH, I had a hard time trying to understand it.

I sincerely doubt that :slightly_smiling_face: It's a trivial `Scanner` goroutine, and `select` below it. The refactor just simplifies the goroutine, and moves the line parsing to the `select`.

I struggled trying to simplify it, as the cyclomatic complexity comes from the `for { select {} }` loop, and the `preferStdErr()` function. The cyclomatic complexity measures the number of possible code paths; not necessarily how difficult the code is to understand. This is subjective, of course, so if you find this version more difficult to understand, and you strongly object `nolint`ing it, then I'll try to refactor it further, with some of your other suggestions.

Otherwise we can revert the refactor and keep the previous version.

Reply:
> I struggled trying to simplify it

I experimented with this and separated the parser code in #619. You can cherry-pick it if you want. This one doesn't have cyclomatic complexity.